src/main/java/uk/gov/hmcts/reform/wataskmanagementapi/services/TaskAutoAssignmentService.java

```diff
@@ -172,15 +149,13 @@ private boolean isRoleAssignmentValid(TaskResource taskResource, Map<String,
     private boolean isTaskRoleNotAutoAssignableOrAuthorisationsNotMatching(RoleAssignment roleAssignment,
                                                                            TaskRoleResource taskRoleResource) {
         return !taskRoleResource.getAutoAssignable()
-               || taskRoleResource.getAutoAssignable()
-                  && isAuthorisationsValid(taskRoleResource)
+               || isAuthorisationsValid(taskRoleResource)
```

Comment:
test coverage?

Reply:
as discussed marking as resolved have already covered tests for all these conditions, since some conditions are unreachable, moreover these is existing code and have improved test coverage